dscache/convert_test.go

```diff
@@ -0,0 +1,327 @@
+		},
+	}
+	dscache := buildDscacheFlatbuffer(userList, dsInfoList)
+	actual := dscache.VerboseString(false)
+
+	expect := `Dscache:
+ Dscache.Users:
+  0) user=test_zero profileID=QmeL2mdVka1eahKENjehK6tBxkkpk5dNQ1qMcgWi7Hrb4B
+  1) user=test_one profileID=QmWYgD49r9HnuXEppQEq1a7SUUryja4QNs9E6XCH2PayCD
+  2) user=test_two profileID=QmPeFTNHcZDr3ZFEfFfepxS5PqHAmfBRGQNPJ389Cwh1as
+ Dscache.Refs:
+  0) initID      = ds_init_id_0000
+     profileID   = QmWYgD49r9HnuXEppQEq1a7SUUryja4QNs9E6XCH2PayCD
+     topIndex    = 0
+     cursorIndex = 0
+     prettyName  = my_ds
+     commitTime  = -62135596800
+     headRef     = /ipfs/QmExampleFirst
+  1) initID      = ds_init_id_0001
+     profileID   = QmWYgD49r9HnuXEppQEq1a7SUUryja4QNs9E6XCH2PayCD
+     topIndex    = 0
+     cursorIndex = 0
+     prettyName  = another_ds
+     commitTime  = -62135596800
+     headRef     = /ipfs/QmExampleSecond
+  2) initID      = ds_init_id_0002
+     profileID   = QmWYgD49r9HnuXEppQEq1a7SUUryja4QNs9E6XCH2PayCD
+     topIndex    = 0
+     cursorIndex = 0
+     prettyName  = checked_out_ds
+     commitTime  = -62135596800
+     headRef     = /ipfs/QmExampleThird
+     fsiPath     = /path/to/workspace/checked_out_ds
+  3) initID      = ds_init_id_0003
+     profileID   = QmPeFTNHcZDr3ZFEfFfepxS5PqHAmfBRGQNPJ389Cwh1as
+     topIndex    = 0
+     cursorIndex = 0
+     prettyName  = foreign_ds
+     commitTime  = -62135596800
+     headRef     = /ipfs/QmExampleFourth
+`
+	if diff := cmp.Diff(expect, actual); diff != "" {
+		t.Errorf("builddscacheFlatbuffer (-want +got):\n%s", diff)
+	}
+}
+
+type DscacheTestRunner struct {
+	prevTsFunc func() int64
+	minute int
+	teardown func()
+}
+
+func NewDscacheTestRunner() *DscacheTestRunner {
+	run := DscacheTestRunner{minute: 0}
+	run.prevTsFunc = logbook.NewTimestamp
+	logbook.NewTimestamp = func() int64 {
+		run.minute++
+		return time.Date(2000, time.January, 1, 0, run.minute, 0, 0, time.UTC).UnixNano()
+	}
+	return &run
+}
+
+func(run *DscacheTestRunner) Delete() {
+	logbook.NewTimestamp = run.prevTsFunc
+	if run.teardown != nil {
+		run.teardown()
+	}
+}
+
+// Test the convertLogbookAndRefs function, which turns a logbook and dsrefs into a list of dsInfo
+func TestConvertLogbookAndRefs(t *testing.T) {
+	run := NewDscacheTestRunner()
+	defer run.Delete()
+
+	ctx := context.Background()
+
+	peerInfo := testPeers.GetTestPeerInfo(0)
+	book := makeFakeLogbook(ctx, t, "test_user", peerInfo.PrivKey)
+
+	dsrefs := []repo.DatasetRef{
+		repo.DatasetRef{
+			Peername: "test_user",
+			ProfileID: profile.ID(peerInfo.PeerID),
+			Name: "first_new_name",
+			FSIPath: "/path/to/first_workspace",
+		},
+		repo.DatasetRef{
+			Peername: "test_user",
+			ProfileID: profile.ID(peerInfo.PeerID),
+			Name: "second_name",
+			FSIPath: "/path/to/second_workspace",
+		},
+	}
+
+	dsInfoList, err := convertLogbookAndRefs(ctx, book, dsrefs)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dsBytes, err := json.MarshalIndent(dsInfoList, "", " ")
```

Comment:
why convert this to JSON? Any reason not to just hand-construct the expected `dsInfoList` & feed that to `cmp.Diff`?

Reply:
I had thought it would be easier to take the json approach, but trying it just now using literals, I was clearly mistaken.